src/test/shell/bazel/bazel_bootstrap_distfile_test.sh

```diff
@@ -114,7 +114,7 @@ function test_bootstrap() {
     # the repo cache is currently missing canonical IDs
     export BAZEL_HTTP_RULES_URLS_AS_DEFAULT_CANONICAL_ID=0
 
-    JAVA_VERSION=exotic # so we don't accidentally match something else
+    JAVA_VERSION=21
```

Comment:
ISTR this being important, can you maybe add an assertion below that we are using the fake toolchain? (perhaps using `aquery` + looking for `dummy.jar`)

Reply:
That's true, but it would be helpful to know what went wrong rather than obscure errors.

Reply:
I found a way to retain the old setup. These checks would have been difficult as the obscure failures would have happened during the loading phase, well before we could have run `aquery`.

Reply:
I found a way to retain the old setup. These checks would have been difficult as the obscure failures would have happened during the loading phase, well before we could have run `aquery`.